scripts/check_style.sh

```diff
@@ -24,17 +24,17 @@ function preparedGrep()
 
 FORMATERROR=$(
 (
-	preparedGrep "#include \"" | egrep -v -e "license.h" -e "BuildInfo.h"  # Use include with <> characters
+	preparedGrep "#include \"" | grep -E -v -e "license.h" -e "BuildInfo.h"  # Use include with <> characters
 	preparedGrep "\<(if|for|while|switch)\(" # no space after "if", "for", "while" or "switch"
 	preparedGrep "\<for\>\s*\([^=]*\>\s:\s.*\)" # no space before range based for-loop
 	preparedGrep "\<if\>\s*\(.*\)\s*\{\s*$" # "{\n" on same line as "if" / "for"
 	preparedGrep "[,\(<]\s*const " # const on left side of type
 	preparedGrep "^\s*(static)?\s*const " # const on left side of type (beginning of line)
 	preparedGrep "^ [^*]|[^*] 	|	 [^*]" # uses spaces for indentation or mixes spaces and tabs
-	preparedGrep "[a-zA-Z0-9_]\s*[&][a-zA-Z_]" | egrep -v "return [&]" # right-aligned reference ampersand (needs to exclude return)
+	preparedGrep "[a-zA-Z0-9_]\s*[&][a-zA-Z_]" | grep -E -v "return [&]" # right-aligned reference ampersand (needs to exclude return)
 	# right-aligned reference pointer star (needs to exclude return and comments)
-	preparedGrep "[a-zA-Z0-9_]\s*[*][a-zA-Z_]" | egrep -v -e "return [*]" -e "^* [*]" -e "^*//.*"
-) | egrep -v -e "^[a-zA-Z\./]*:[0-9]*:\s*\/(\/|\*)" -e "^test/"
+	preparedGrep "[a-zA-Z0-9_]\s*[*][a-zA-Z_]" | grep -E -v -e "return [*]" -e "^* [*]" -e "^*//.*"
+) | grep -E -v -e "^[a-zA-Z\./]*:[0-9]*:\s*\/(\/|\*)" -e "^test/"
```

Comment:
Does this work on MacOS as well? I think we had some problems in the past because of `grep -E`, but might be imagining that

Reply:
I haven't checked that yet. This is the reason why this and #10585 are still drafts - I need to test this stuff a bit more because we have very little coverage for shell scripts but I want to finish the bytecode compare PRs first.

Reply:
This should be fine on macOS. `-E` is one of the options that work the same way between Linux and macOS. AFAIK `-P` (`--perl-regexp`) that we're using in `get_version.sh` is the problematic one.